### PR TITLE
[18CZ] Added 2 player variant

### PR DESCRIPTION
**Pull request by Zwergenpunk**
Adds the 2 player variant. The variant is activated automatically when 2 player are selected

### CODE REVIEW

File: lib/engine/game/g_18_cz/round/operating.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      module Round
+        class Operating < Engine::Round::Operating
+          def after_process(action)
+            return super if @game.multiplayer?
+
+            entity = @entities[@entity_index]
+            if @game.corporation_of_vaclav?(entity)
+              case action
+              when Engine::Action::RunRoutes
+                process_action(Engine::Action::Dividend.new(entity, kind: 'payout')) if action.routes.any?
```
Comment (tobymao) [inline lib/engine/game/g_18_cz/round/operating.rb:17]: unless empty

File: lib/engine/game/g_18_cz/game.rb [hunk 2]
```diff
@@ -2498,35 +2508,127 @@ def setup
...
+
+            @corporations.select { |item| item.type == :large }.each { |item| item.max_ownership_percent = 70 }
+
+            @players << @vaclav
+          end
 
           # Only small companies are available until later phases
           @corporations, @future_corporations = @corporations.partition { |corporation| corporation.type == :small }
 
+          new_corporation_for_vaclave(:small) unless multiplayer?
```
Comment (yzemaze) [inline lib/engine/game/g_18_cz/game.rb:2525]: typo vaclav**e**?

Comment (Zwergenpunk) [reply lib/engine/game/g_18_cz/game.rb:2525]: Yeah. Thanks

File: lib/engine/game/g_18_cz/game.rb [hunk 3]
```diff
@@ -2498,35 +2508,127 @@ def setup
...
+          active.reject { |item| item == @vaclav }
+        end
+
+        def optional_hexes
+          return game_hexes if multiplayer?
+
+          new_hexes = {}
+          game_hexes.keys.each do |color|
+            new_map = game_hexes[color].map do |coords, tile_string|
+              [coords - TWO_PLAYER_HEXES_TO_REMOVE, tile_string]
```
Comment (yzemaze) [inline lib/engine/game/g_18_cz/game.rb:2595]: .transform_keys
cf. https://github.com/tobymao/18xx/pull/4653/commits/e5b152864eb10c368be4769220bb236e50d72104

File: lib/engine/game/g_18_cz/game.rb [hunk 4]
```diff
@@ -2498,35 +2508,127 @@ def setup
...
+            end.to_h
+            new_hexes[color] = new_map
+          end
+          new_hexes
+        end
+
+        def multiplayer?
+          @multiplayer ||= @players.reject { |item| item == @vaclav }.size >= 3
```
Comment (yzemaze) [inline lib/engine/game/g_18_cz/game.rb:2603]: .count
cf. https://github.com/tobymao/18xx/pull/4653/commits/c16c9642205aed389b2a27fae09edb9979cb85c5

Comment (Zwergenpunk) [reply lib/engine/game/g_18_cz/game.rb:2603]: oh yeah. I had >= 3 first and the added the reject but did not change the number

File: lib/engine/game/g_18_cz/game.rb (hunk 3 above)
Comment (Zwergenpunk) [reply lib/engine/game/g_18_cz/game.rb]: thanks. did not know that even exists ;)

**Comment by Zwergenpunk** [comment]
Also added info to welcome page

**Comment by michaeljb** [comment]
Testing in a local multiplayer game:
* the non-priority player was given control during the track laying step
* for the run routes step, it would probably be nice to match how 1846 receivership works, where the corp isn't owned by any particular player and all players are alerted as active
* it would be ideal for the "Next SR" counts to not include Vaclav
* since you assign ownership of Vaclav's companies, looking at the Entities tab during ORs lists Vaclav's companies under the player instead of under Vaclav

**Comment by Zwergenpunk** [comment]
Ah now I understand how the active_players work. Thanks alot.
I did not read the rules correct. I just assumed track lay and token placment is always the same and not depending on the size of corporation.

**Comment by Zwergenpunk** [comment]
Removed Vaclav from Priority Informations

**Pull request merged**